Quotient/connection.h

```diff
@@ -760,6 +760,9 @@ public Q_SLOTS:
     static Connection* makeMockConnection(const QString& mxId,
                                           bool enableEncryption = E2EE_Enabled);
 
+    static Connection* makeMockConnection(Connection *connection, const QString& mxId,
```

Comment:
this API looks weird, why is it a static function that's passed an object?

Looking at the neochat MR for this, it seems that we could easily use the existing makeMockConnection function

Reply:
the issue with the existing function is that it return a `Connection *` and not a `NeoChatConnection *`

Reply:
`AccountRegistry` looks like a good place for it? Then you could derive from it and override a virtual function. Or simply making `makeMockConnection()` a template would also work I guess?